packages/nuxt/src/app/composables/hydrate.ts

```diff
@@ -23,3 +23,21 @@ export const useHydration = <K extends keyof NuxtPayload, T = NuxtPayload[K]> (k
     })
   }
 }
+
+/**
+ * A `requestIdleCallback` options utility, used to determine custom timeout for idle-callback based delayed hydration.
+ * @param opts the options object, containing the wanted timeout
+ */
```

Comment:
As previous comment, this only makes sense if they return a promise.

Reply:
I don't think I understand why it needs to be a promise? Why can it not be a synchronous override?

Reply:
from a naming point of view* i'd go for something with a define prefix if it's just for the types

Reply:
Renamed to `hydrate`